cpp/src/parquet/statistics.cc

```diff
@@ -543,7 +557,7 @@ class TypedStatisticsImpl : public TypedStatistics<DType> {
     ResetCounts();
     has_min_max_ = false;
     has_distinct_count_ = false;
-    has_null_count_ = false;
+    has_null_count_ = true;
```

Comment:
page statistics will call `Reset`, this will clear `has_null_count_`, however, this time it should have `has_null_count_ = 0`

Reply:
What about adding something like `ResetHasFlags()` below?
```cpp
void ResetHasFlags() {
  has_min_max_ = false;
  has_distinct_count_ = false;
  has_null_count_ = true;
}
```
We can call it in the constructors and Reset() to make sure the initial states are consistent.

Reply:
Updated